chapters/en/slides/module5/module5_03.Rmd

```diff
@@ -0,0 +1,316 @@
+---
+params:
+  dynamictitle: "module5_03"
+title: "`r params$dynamictitle`"
+output:
+  md_document:
+    variant: gfm
+    pandoc_args: "--markdown-headings=atx"
+---
+
+```{r setup, include=FALSE}
+source('../../../../slide-setup.R')
+# Remember to also manually update the YAML title above
+```
+
+type: slides
+
+
+# Defining and transforming axis ranges
+
+Notes:
+In this slide deck,
+we will see how defining an explicit axis range
+is beneficial when we need to zoom into a region of a plot,
+and how log transforming an axis can help increase resolution of small values.
+
+---
+
+## Read in the data containing the daily sum of donations to Wikipedia in 2020
+
+```{python}
+# TODO save wrangled data?
+import pandas as pd
+import altair as alt
+from vega_datasets import data
+
+donations = (
+    pd.read_csv("https://frdata.wikimedia.org/donationdata-vs-day.csv", parse_dates=['date'])
+    .query('2020 < date < 2021')
+    .assign(week_day=lambda df: df['date'].dt.day_name().str[:3]))
+
+alt.Chart(donations, title='Wikipedia donations peak during Christmas').mark_circle().encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day'))
+```
+
+Notes:
+We will be working with a data set that contains the daily sum
+of all donations received by Wikipedia
+in the year 2020.
+
+From this overview of the data
+we can see that the most money
+is donated around Christmas and new years.
+
+There also seems to be a higher amount donated during the fall
+than during spring and winter.
+
+However,
+we cannot see exactly which values the data points during the first half of the year has.
+Are they close to zero? Or 10,000? We can't know from looking at this plot.
+
+---
+
+## We can zoom in to highlight a region of an axes by filtering the data
+
+```{python}
+title_text = 'Wikipedia donations peak during Christmas'
+alt.Chart(donations[donations['date'].between("2020", "2020-07")], title=title_text).mark_circle().encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day'))
+```
+
+Notes:
+Zooming in on an axis is useful
+when we want to highlight a certain region.
+The easiest way to achieve this
+is usually to filter the data we pass to `Chart` as in this example.
+
+---
+
+## We can zoom in to highlight a region of an axes
+
+```{python}
+title_text = 'Wikipedia donations peak during Christmas'
+alt.Chart(donations[donations['date'].between("2020", "2020-07")], title=title_text).mark_circle(clip=True).encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day', scale=alt.Scale(domain=[0, 100_000])))
```

Comment:
This line is too long, can  we split it up?

Reply:
There are others in this slide deck that are also on the almost too long side... please take a look at all the line lengths viewing the slides on a standard laptop.